### PR TITLE
Release candidate 2021 H1

### LINKED ISSUE #97: Add OpenDP as a ref and resource on diff priv
[OpenDP](https://projects.iq.harvard.edu/opendp/about):

> OpenDP is a community effort to build a trustworthy suite of open-source tools for enabling privacy-protective analysis of sensitive personal data, focused on a library of algorithms for generating differentially private statistical releases.   The target use cases for OpenDP are to enable government, industry, and academic institutions to safely and confidently share sensitive data to support scientifically oriented research and exploration in the public interest.  We aim for OpenDP to flexibly grow with the rapidly advancing science of differential privacy, and be a pathway to bring the newest algorithmic developments to a wide array of practitioners.

> OpenDP is led by Faculty Directors Gary King and Salil Vadhan and an Executive Committee at Harvard University, funded by a grant from the Sloan Foundation.   Its efforts so far have included implementing a differentially private curator application in collaboration with Microsoft, and developing a framework for a community-governed OpenDP Commons through the work of an Ad Hoc Design Committee including external experts.  Going forward, the project plans to engage with a wide community of stakeholders, establish partnerships with a variety of groups from industry, academia, and government, and eventually transition to full community governance with a formal steering committee.

### LINKED ISSUE #99: Add HRX's article into the "various controversies" section
Article about a combinaison of some bias and human rights non conformities regarding the construction and use of the CONARC database and the Facial Recognition System for Fugitives in Buenos Aires:

https://www.hrw.org/news/2020/10/09/argentina-child-suspects-private-data-published-online

### LINKED ISSUE #103: Add interesting references
- [En route vers le cycle de vie des modèles !](https://www.quantmetry.com/blog/premier-etape-cycle-vie-modeles/), Quantmetry, Janvier 2020
- [MLflow](https://mlflow.org/): *MLflow is an open source platform to manage the ML lifecycle, including experimentation, reproducibility, deployment, and a central model registry*
- [Pandas Profiling](https://github.com/pandas-profiling/pandas-profiling) : sur la reco de Nathan
- [Minneapolis prohibits use of facial recognition software by its police department](https://www.theverge.com/2021/2/13/22281523/minneapolis-prohibits-facial-recognition-software-police-privacy)

### LINKED ISSUE #106: Add interesting reference: factsheet by IBM
[J’ai une ressource supplémentaire à proposer] de Emilie Sirvent-Hien, emilie.hien@orange.com sur l'objet Q4.2 Conditions et limites d'utilisation d'un modèle.

Message de l'utilisateur : Bonjour, Vous pourriez citer également la proposition IBM avec les factsheet qui permet de documenter les modèles d'IA et informer sur les limites, cf: https://aifs360.mybluemix.net/

### LINKED ISSUE #108: Add an answer to Q4.3
Feedback sur Q4.3 Analyse et partage d'incidents (master_section id=187)
[Aucun élément de réponse ne me convient] de Daniel Bartolo, daniel.bartolo@externe.maif.fr sur l'objet Q4.3 Analyse et partage d'incidents.

Message de l'utilisateur : Il manque un élément de réponse : tritement et analyse mais pas de publication ou de partage externe sur le sujet

### LINKED ISSUE #109: Set an explanation to each evaluation element
For the web platform, in order to manage the translations, the field "explanation" is marked as a "translation field". 
This leads to the fact that an explanation is required for each evaluation element to validate a language, it cannot be empty/None.

However, the evaluation element Q2.2 has no explanation in both languages. So it results in the non-validation of the languages. 
For now, a temporary solution has been implemented to do not take this field into consideration but this is much easier & more robust to add an explanation to this evaluation element, regardless the content.

### LINKED ISSUE #112: Add Shapash and FACET in resources

### LINKED ISSUE #114: Add DataforGood ressources

### LINKED ISSUE #116: Références : The Global Landscape of AI Guidelines & Code Carbon
Récemment, 2 sujets m'ont semblé intéressants à vous partager : 

1. [The Global Landscape of AI Guidelines](https://www.nature.com/articles/s42256-019-0088-2.epdf)

Publié par Anna Jobin, qui travaille sur les dimensions sociales des systèmes algorithmiques depuis Zürich. (Elle donne également son [point de vue](https://googlewalkout.medium.com/standing-with-dr-timnit-gebru-isupporttimnit-believeblackwomen-6dadc300d382) sur le licenciement récent de Timnit Gebru, ex-directrice de la recherche sur l'éthique chez Google AI et de ses conséquences sur la discipline).

Vous pouvez trouver [des notes](https://docs.google.com/document/d/1iPwGoTPa0Gz70YUJDgXmI3oFE7IDw_8_pb9Hy56xTtM/edit?usp=sharing) sur la présentation de l'article par Anna Jobin effectuée dans le cadre du séminaire de l'ecn-EHESS "Étudier les cultures numériques" donné par Antonio Casilli, le 27 janvier dernier.

2. [Code Carbon](https://codecarbon.io/)

Le travail effectué par notamment Yoshua Bengio sur la mise en application de certains principes évoqués dans le référentiel dans l'outil [Code Carbone](https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&arnumber=9325146). L'outil présente quelques exemples sur sa [documentation sommaire](https://mlco2.github.io/codecarbon/). Il a également été présenté à la conférence SocML202 et des notes des présentations sont disponibles [par ici](https://docs.google.com/document/d/1RgLNfJOLWOC4QJ7g3YbAVT-JxAS001_pJ3C660AFvVo/edit?usp=sharing).

### LINKED ISSUE #118: Resource - dead link
For the resource [Technical guide] - Privacy Enhancing Technologies Decision Tree (v2), Private AI, 2020, displayed in the Q1.7 and Q1.8, the link is dead and there is a 404 issue.

So either find the new link and remove this resource.

Note that I created an issue for the platform to report automatically the "dead link", which can unfortunately happen often.

### LINKED ISSUE #119: Missing blank lines between section titles and keywords
For the sections 1 and 2 of the English markdown assessment, there is a missing `\n` between the section title and the keyword in the raw file - but it is correctly displayed in markdown file.

`### Section 1 - Protecting personal or confidential data
**[Data privacy]**`

should be: 

`### Section 1 - Protecting personal or confidential data

**[Data privacy]**`

### LINKED ISSUE #120: Reference: Principled Artificial Intelligence | Berkman Klein Center (harvard.edu)
Study and add meta study: Principled Artificial Intelligence | Berkman Klein Center (harvard.edu)

### LINKED ISSUE #123: AI registers
Check how to integrate AI registers:
https://towardsdatascience.com/ai-registers-finally-a-tool-to-increase-transparency-in-ai-ml-f5694b1e317d

### LINKED ISSUE #133: A bunch of suggestions
From @JustineBoulant:

- [x] Update the README: Make it clearer that the output of this participative work is a reference framework
- [x] Update the README: Clarify the sentence on the scope of existing guidelines/declarations/charters
- [ ] Scoping and naming: revive the data science vs. AI debate. Should it include econometry?
- [x] Typos: "inititives" in `references.md`, "est le fruit travail" in `referentiel_evaluation.md`
- [x] Clarify the sentence « Il y a cependant très peu de choses sur comment un modèle est élaboré » in `references.md`
- [x] Enhance element on minimisation principle / privacy by design with alternatives like correspondence tables and data dispatching
- [x] Enhance explanations/context infos on ML vulnerabilities (elements 1.7 and 1.8)
- [ ] Enhance climate impact / CO2 aspects: new answer items? additional evaluation element?

### LINKED ISSUE #134: ML Exploit : Remote code execution from pickle files
See this [reference](https://blog.trailofbits.com/2021/03/15/never-a-dill-moment-exploiting-machine-learning-pickle-files/) for details : 
- Pickle format allows python code execution ([Proof of concept](https://github.com/trailofbits/fickling/blob/master/example/pytorch_poc.py))
- PyTorch Hub official response : this security aspect shouldn't be responsibility of a model hosting platform.

I guess PyTorch official comment could have been a bit more responsible.
Linked with 1.7/1.8/1.9, 4.4, 5.2

### LINKED ISSUE #136: Moving beyond "algorithmic bias is a data problem"
From @RomainGoussault:
> [Moving beyond "algorithmic bias is a data problem"](https://www.cell.com/patterns/fulltext/S2666-3899(21)00061-1)

Question it raises: should we add an evaluation element (2.4 ?) on this aspect?

### LINKED ISSUE #140: [User feedback on the assessment] Missing answer item
[Aucun élément de réponse ne me convient] de Mohainfo Lafier, chetbi1997@gmail.com sur l'objet Q1.4 Législation et exigences contractuelles applicables - Audit et certification.

### LINKED ISSUE #144: Add Counterfit to resources on ML security

### PR DESCRIPTION
**Pull request by bowni**
### Resources/references to add

- [x] Fixes #97 : OpenDP on differential privacy
- [x] Fixes #99 : HRX article on AI use cases controversies (via @meuce)
- [x] Fixes #103 : Misc. references
- [x] Fixes #106 : IBM factsheet
- [x] Fixes #112 : Shapash & FACET on explainabillity
- [x] Fixes #114 : Resources elaborated during the Dataforgood season 8 project (fairness, genealogy, robustness)
- [x] Fixes #116 : Code Carbon (via @SaboniAmine)
- [x] Fixes #120 : Berkman meta study on AI ethics principles
- [x] Fixes #134 : ML Exploit with pickle files
- [x] Fixes #144 : Counterfit to test ML models vulnerabilities

### Changes in the evaluation elements and answer items

- [x] Fixes #108 : New answer item to Q4.3 on sharing/publishing AI incidents (via D. Bartolo)
- [x] Fixes #123 : Public AI Registers (new reference + new answer item to 5.5)
- [x] Fixes #136 : New evaluation element on modeling- and learning-related biases
- [x] Fixes #140 : New answer item to 1.4 on compliance to personal data regulations (user feedback)
- [x] Fixes #133 : New answer item to Q1.5 on alternatives to minimisation principle (via @JustineBoulant) 

### Misc. fixes

- [x] Fixes #109 : Add missing explanations/hints in some elements
- [x] Fixes #118 : Broken link
- [x] Fixes #119 : Missing newlines
- [x] Fixes #133 : Clarify README and `References.md`, fix typos, enhance explanations/context infos on ML vulnerabilities in Q1.7-1.8)

### CODE REVIEW

**Review by ClementMayer — APPROVED** [review]

File: assessment_framework_eng.md [hunk 1]
```diff
@@ -379,6 +385,43 @@ Complement on the use of synthetic data and _data augmentation_, _re-weighting_
 
 </details>
 
+---
+
+Q2.4 : **Links between modelisation choices and bias**  
+_(Condition : R2.2 <> 2.2.b)_  
+Recent work has shown the role that modeling and learning choices can play in the formation of discriminatory bias. Differential privacy, compression, the choice of the learning rate, early stopping mechanisms for example can have disproportionate impacts on certain subgroups. Within your organisation, the general level of knowledge of collaborators working on data science projects on this topic is:
```
[the remark this answers is not in the document]
Comment (bowni) [reply assessment_framework_eng.md:392]: Excellent, thanks @SaboniAmine, I'll add it!

**Comment by bowni** [comment]
From @meuce : on answer item 1.4.d it should read "not **at** the organization level". I'll do the edit!

File: assessment_framework_eng.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (bowni) [reply assessment_framework_eng.md:392]: Done

File: assessment_framework_eng.md [hunk 2]
```diff
@@ -221,7 +232,7 @@ _(Select all the answer items that correspond to practices in your organisation)
 <details>
 <summary>Expl1.8 :</summary>
 
-The state of the art in ML security is constantly evolving. While it is impossible to guard against all vulnerabilities at all times, it is crucial to be aware of them and to keep a watch on them. The article [Demystifying the Membership Inference Attack](https://medium.com/disaitek/demystifying-the-membership-inference-attack-e33e510a0c39) is for example an interesting entry point in the context of sensitive data.
+TThe state of the art in ML security is constantly evolving. If data scientists are now familiar in general with the membership inference attack (see proposed resources), new ones are being published regularly. While it is impossible to guard against all vulnerabilities at all times, it is crucial to be aware of them and to keep a watch on them. The article [Demystifying the Membership Inference Attack](https://medium.com/disaitek/demystifying-the-membership-inference-attack-e33e510a0c39) is for example an interesting entry point in the context of sensitive data.
```
Comment (celinejacques) [inline assessment_framework_eng.md:235]: There is an extra T at the beginning of the sentence

File: assessment_framework_eng.md [hunk 3]
```diff
@@ -302,6 +317,7 @@ It is a question of ensuring that oneself considers these subjects and therefore
 
 - (Web article) *[Hidden Bias](https://pair.withgoogle.com/explorables/hidden-bias/)* explorable from [PAIR](https://pair.withgoogle.com/)
 - (Technical guide) *[Tour of Data Sampling Methods for Imbalanced Classification](https://machinelearningmastery.com/data-sampling-methods-for-imbalanced-classification/)*
+- (Software & Tools) *[Pandas Profiling](https://github.com/pandas-profiling/pandas-profiling)*: *Create HTML profiling reports from pandas `DataFrame` objects. The pandas `df.describe()` function is great but a little basic for serious exploratory data analysis. `pandas_profiling` extends the pandas `DataFrame` with `df.profile_report()` for quick data analysis
```
Comment (celinejacques) [inline assessment_framework_eng.md:320]: There is an extra * before Create

File: assessment_framework_eng.md [hunk 4]
```diff
@@ -869,6 +933,8 @@ Technical resources such as SHAP or LIME provide a first-hand introduction to th
 - (Technical guide) *[Interpretable Machine Learning, A Guide for Making Black Box Models Explainable](https://christophm.github.io/interpretable-ml-book/)*, Christoph Molnar
 - (Web article) *[Understanding model predictions with LIME](https://towardsdatascience.com/understanding-model-predictions-with-lime-a582fdff3a3b)*, blog L. Hulstaert, 2018
 - (Software & Tools) *[SHAP](https://github.com/slundberg/shap): A game theoretic approach to explain the output of any machine learning model*.
+- (Software & Tools) *[Shapash](https://github.com/MAIF/shapash): a MAIF Datalab project which aims to make machine learning interpretable and understandable by everyone. It provides several types of visualization that display explicit labels that everyone can understand
```
Comment (celinejacques) [inline assessment_framework_eng.md:936]: It misses a * here

File: references.md [hunk 5]
```diff
@@ -37,11 +37,15 @@
 
   - La *distillation* d'un modèle, en plus de la compression qu'elle apporte, peut être utilisée comme une mesure de protection du modèle et des données d'entraînement utilisées, voir par exemple *[Knowledge Distillation : Simplified](https://towardsdatascience.com/knowledge-distillation-simplified-dd4973dbc764)*, Towards Data Science, 2019, et *[Distilling the Knowledge in a Neural Network](https://arxiv.org/abs/1503.02531)*, G. Hinton, O. Vinyals, J. Dean, 2015
 
+- Cycle de vie complet :
+
+  - **[En route vers le cycle de vie des modèles !](https://www.quantmetry.com/blog/premier-etape-cycle-vie-modeles/)*, G. Martinon, Janvier 2020
```
Comment (celinejacques) [inline references.md:42]: There is an extra * here

File: referentiel_evaluation.md [hunk 6]
```diff
@@ -304,6 +317,7 @@ Il s'agit de s'obliger à s'interroger sur ces sujets et donc à réfléchir aux
 
 - (Web article) *[Hidden Bias](https://pair.withgoogle.com/explorables/hidden-bias/)* explorable from [PAIR](https://pair.withgoogle.com/)
 - (Technical guide) *[Tour of Data Sampling Methods for Imbalanced Classification](https://machinelearningmastery.com/data-sampling-methods-for-imbalanced-classification/)*
+- (Software & Tools) *[Pandas Profiling](https://github.com/pandas-profiling/pandas-profiling)*: *Create HTML profiling reports from pandas `DataFrame` objects. The pandas `df.describe()` function is great but a little basic for serious exploratory data analysis. `pandas_profiling` extends the pandas `DataFrame` with `df.profile_report()` for quick data analysis
```
Comment (celinejacques) [inline referentiel_evaluation.md:320]: Il y a une * en trop avant Create

File: referentiel_evaluation.md [hunk 7]
```diff
@@ -867,6 +933,8 @@ Des ressources techniques comme SHAP ou LIME permettent d'entrer de plain-pied d
 - (Technical guide) *[Interpretable Machine Learning, A Guide for Making Black Box Models Explainable](https://christophm.github.io/interpretable-ml-book/)*, Christoph Molnar
 - (Web article) *[Understanding model predictions with LIME](https://towardsdatascience.com/understanding-model-predictions-with-lime-a582fdff3a3b)*, blog L. Hulstaert, 2018
 - (Software & Tools) *[SHAP](https://github.com/slundberg/shap): A game theoretic approach to explain the output of any machine learning model*
+- (Software & Tools) *[Shapash](https://github.com/MAIF/shapash): un projet open source de MAIF Datalab facilitant la prise en main et permettant de visualiser les analyses d'explicabilité et d'interprétabilité des modèles
```
Comment (celinejacques) [inline referentiel_evaluation.md:936]: Il manque une *

**Review by celinejacques — APPROVED** [review]

**Comment by bowni** [comment]
Thank you @celinejacques - will fix those typos right away!

**Comment by bowni** [comment]
@celinejacques fixing the typos and committing it dismissed automatically your review - could you reiterate it please?

File: assessment_framework_eng.md [hunk 8]
```diff
@@ -358,6 +381,45 @@ Complement on the use of synthetic data and _data augmentation_, _re-weighting_
...
+
+- [ ] 2.4.a Complete beginner
+- [ ] 2.4.b Basic
+- [ ] 2.4.c Confirmed
+- [ ] 2.4.d Expert
+
+<details>
+<summary>Expl2.4 :</summary>
+
+If datasets used to train and evaluate a model require a particular attention to prevent discriminatory biases, recent work show that modeling choices have to be taken into account too. The article *"Moving beyond “algorithmic bias is a data problem”"* suggested in resources synthesizes very well how the learning algorithm, the model structure, adding or not differential privacy, compression, etc. can have consequences on the fairness of a model. Extracts:
```
Comment (SaboniAmine) [inline assessment_framework_eng.md:406]: Typo: Recent work shows / have shown.

**Review by SaboniAmine — DISMISSED** [review]

**Review by SaboniAmine — APPROVED** [review]

**Review by celinejacques — APPROVED** [review]

**Comment by bowni** [comment]
👍

**Pull request merged**